### PR TITLE
feat: [CDS-41634]:skip-instances

### PR DESCRIPTION
**Pull request by litvinovstas**
### Summary

Improves after demo feedback
Add skip instances checkbox for ssh/winrm,
Moved from left to right side PackageType and Phases fields for Execution strategy ssh/winrm,
Changed default value for PacjageType from WAR to JAR,
Remove Phase field for Rolling strategy,
resolved issue with display file location on config file view list.

#### Screenshots

NA
Screenshot 2022-08-26 at 12 06 57
Screenshot 2022-08-26 at 12 09 47
Screenshot 2022-08-26 at 12 10 19

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 14 Issues 
* Code issue 14 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10688)

File: src/modules/70-pipeline/components/PipelineStudio/ExecutionStrategy/Phases.tsx [hunk 1]
```diff
@@ -60,6 +61,14 @@ enum ExecutionType {
 }
 
 const packageTypeItems = [
+  {
+    label: 'JAR',
```
Comment (bhavya-sinha) [inline src/modules/70-pipeline/components/PipelineStudio/ExecutionStrategy/Phases.tsx:65]: Can we add Stringkeys here from strings file?

File: src/modules/70-pipeline/components/PipelineStudio/ExecutionStrategy/Phases.tsx [hunk 2]
```diff
@@ -325,79 +327,85 @@ function Phases({ selectedStrategy, serviceDefinitionType, selectedStage }: Phas
...
-                                      name={`phases[${index}]`}
-                                      label={getString('common.instanceLabel')}
-                                      readonly={false}
-                                      onChange={value => {
-                                        const currentValue = get(formikProps?.values, `phases[${index}]`)
+                          <div className={css.phasesListContainer}>
+                            {Array.isArray(formikProps.values.phases) &&
+                              formikProps.values.phases.map((field: InstanceFieldValue, index: number) => {
+                                return (
+                                  <Layout.Vertical key={index}>
```
Comment (bhavya-sinha) [inline src/modules/70-pipeline/components/PipelineStudio/ExecutionStrategy/Phases.tsx:334]: Avoid using array map index as React keys. Use some unique key instead

File: src/modules/70-pipeline/components/PipelineStudio/ExecutionStrategy/Phases.tsx [hunk 3]
```diff
@@ -325,79 +327,85 @@ function Phases({ selectedStrategy, serviceDefinitionType, selectedStage }: Phas
...
+                                    <Layout.Horizontal
+                                      key={index}
```
Comment (bhavya-sinha) [inline src/modules/70-pipeline/components/PipelineStudio/ExecutionStrategy/Phases.tsx:336]: index not required here

File: src/modules/70-pipeline/components/PipelineStudio/ExecutionStrategy/Phases.tsx [hunk 4]
```diff
@@ -325,79 +327,85 @@ function Phases({ selectedStrategy, serviceDefinitionType, selectedStage }: Phas
...
-                                    {isCanary ? (
-                                      <Container className={css.removePhase}>
-                                        <Button
-                                          icon="main-trash"
-                                          iconProps={{ size: 20 }}
-                                          minimal
-                                          data-testid={`remove-phases-[${index}]`}
-                                          onClick={() => remove(index)}
-                                          disabled={loading || formikProps.values.phases.length <= 1}
+                                          if (currentValue.type !== value.type) {
```
Comment (bhavya-sinha) [inline src/modules/70-pipeline/components/PipelineStudio/ExecutionStrategy/Phases.tsx:362]: Can you please onchange to a separate func.

File: src/modules/70-pipeline/components/PipelineStudio/ExecutionStrategy/Phases.tsx [hunk 5]
```diff
@@ -325,79 +327,85 @@ function Phases({ selectedStrategy, serviceDefinitionType, selectedStage }: Phas
...
+                                                        percentage: currentValue?.spec?.percentage || 1
+                                                      }
+                                              }))
+                                            )
+                                            return
+                                          }
+                                          formikProps.setFieldValue(`phases[${index}]`, { ...value })
+                                        }}
+                                        expressions={expressions}
+                                        allowableTypes={[MultiTypeInputType.FIXED]}
```
Comment (bhavya-sinha) [inline src/modules/70-pipeline/components/PipelineStudio/ExecutionStrategy/Phases.tsx:382]: If allowable types has just fixed option, can we use dropdown instead

File: src/modules/70-pipeline/components/PipelineStudio/SkipInstances/SkipInstances.tsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  selectedStage?: SkipInstancesConfig
+  isReadonly: boolean
+  onUpdate(check: boolean): void
+}
+
+function SkipInstances(props: SkipInstancesProps): JSX.Element {
+  const { getString } = useStrings()
+  const { isReadonly, onUpdate, selectedStage } = props
+  return (
+    <>
```
Comment (bhavya-sinha) [inline src/modules/70-pipeline/components/PipelineStudio/SkipInstances/SkipInstances.tsx:27]: Fragment not needed here

File: src/modules/75-cd/components/PipelineStudio/DeployAdvancedSpecifications/DeployAdvancedSpecifications.tsx [hunk 7]
```diff
@@ -188,6 +199,31 @@ const DeployAdvancedSpecifications: React.FC<AdvancedSpecifications> = ({ childr
...
         </Card>
+        {SSH_NG && getSshOrWinRmType() ? (
+          <div data-testid="skip-instances" className={stageCss.tabHeading}>
+            <SkipInstances
+              selectedStage={stage?.stage}
+              isReadonly={isReadonly}
+              onUpdate={checked => {
+                const { stage: pipelineStage } = getStageFromPipeline(selectedStageId)
+
+                if (pipelineStage && pipelineStage.stage) {
```
Comment (bhavya-sinha) [inline src/modules/75-cd/components/PipelineStudio/DeployAdvancedSpecifications/DeployAdvancedSpecifications.tsx:210]: We can use optional chaining instead `pipelineStage?.stage`

**Review by bhavya-sinha — APPROVED** [review]

**Review by YounixIO — APPROVED** [review]

**Pull request merged**